saleor/giftcard/gateway.py

```diff
@@ -254,9 +256,107 @@ def charge_gift_card_transactions(
                 gift_card.save(update_fields=["current_balance_amount"])
                 response["result"] = TransactionEventType.CHARGE_SUCCESS.upper()
                 response["message"] = f"Gift card (ending: {gift_card.display_code})."
+                response["actions"] = [TransactionAction.REFUND.upper()]
 
             create_transaction_event_from_request_and_webhook_response(
                 transaction_event,
                 None,
                 transaction_webhook_response=response,
             )
+
+
+def cancel_gift_card_authorization(
```

Comment:
It is only sugestion so it is up to you, but I would recommend name like: `cancel_gift_card_transaction`.
For me It is obvious that `cancel -> authorization` and `refund -> charge`.  You can't make `cancel` when your funds are charged. This is why I would drop the `authorization` and `charge` (from the same method but for refund).
Adding `transaction` confirms me that we're "inside" TransactionAPI scope